cognite/client/_api/ai/tools/documents.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes.ai import Answer, AnswerLanguage, Summary
+from cognite.client.data_classes.data_modeling import NodeId
+from cognite.client.utils.useful_types import SequenceNotStr
+
+
+class AIDocumentsAPI(APIClient):
+    _RESOURCE_PATH = "/ai/tools/documents"
+
+    def summarize(
+        self,
+        ids: Sequence[int] | None = None,
+        external_ids: SequenceNotStr[str] | None = None,
+        instance_ids: Sequence[NodeId] | None = None,
+        ignore_unknown_ids: bool = False,
+    ) -> list[Summary]:
```

Comment:
Let's make a `SummaryList`

Reply:
Could you shed some light on why that is useful? I come from Go and have learned the hard way to prefer primitive types, but I see it's very common in the sdk

Reply:
Generally I agree with you, but users of the SDK expect certain features from our "list of resource" classes such as:
- `get` method for hash table backed item lookup
- pretty-printing in notebooks
- easy conversion to pandas data frame
- load and dump methods to/from dict and yaml
...all of which come for free by inheriting from the base Cognite list class